main.py

```diff
@@ -46,39 +47,148 @@
 import argparse
 import re
 import tuf
+import hashlib
+import json
 
 TOTO_TOOL_VERSION = "Toto Build/Test Metadata Generator 0.4"
 DEFAULT_POLICY_FILENAME = "default_policy.json"
+DEFAULT_CUR_DIR = os.getcwd()
+DEFAULT_WORK_DIR = os.path.join(DEFAULT_CUR_DIR, "work")
 
 def main():
+  # Setup the command line parser
+  args = get_command_line_args()
+
+  # Read the commands and begin execution
+  read_command_policy(args)
+
+
+def read_command_policy(args):
   """
   <Purpose>
-    Straightforward function to encapsulate the program's core logic. Called
-    at the end of this file.
+    This function reads and sets up the commands that will be executed.
+    Commands are either passed in:  1) via the command_policy file option
+    or 2) a single command in the COMMAND option.  The command_policy 
+    option will be more robust (ex: allows users to set the cmd_name, 
+    set a verify_command) while the single command omits these options.
+    Since the single command option does not allow a cmd_name to be 
+    specified, the output files for this option are prepended with
+    "generic" (ex: generic_metadata.json).
+ 
+    This function will create a list of executing commands and their 
+    attributes (cmd_name, verify_command, input_file), validate 
+    attributes, execute the command.  Take the hash value from the 
+    metatdata and create a cumulative hash.  These values are then 
+    stored into a main_metadata.json file.
+   
+    Note:  function asserts that cmd_name must be filled out 
+    for the command_policy option and if an input file is 
+    specified then it must exist else the program will exit.  
 
   <Arguments>
-    None.
+    args:
+      The arguments that are passed from the command line.
 
   <Exceptions>
-    TBD.
+    IOError: 
+      Returned if the input file does not exist.
 
-  <Return>
+  <Returns>
     None.
   """
 
-  # Setup the command line parser
-  args = get_command_line_args()
+  commands = []
+  # Setup the command policy here.  The user can either pass
+  # a file in from the commandline (command_policy option), 
+  # otherwise pass a single command line entry.
+  if args.command_policy:
+    # Begin processing the commands from the command policy. 
+    command_policy_filepath = args.command_policy
+    commandList = tuf.util.load_json_file(command_policy_filepath)
+    commands = commandList["commands"]
+  else:
+    # The command is read from the commandline (COMMAND var).
+    # We will autoname the file generic_metadata.json. 
+    tmp_commands = [[args.command, None, "generic", args.input]]
+    commands = json.dumps(tmp_commands)
```

Comment:
I agree since it is not fully completed yet (tarball hashes not implemented
yet). Although we can still talk about it/demo during the presentation.

On Sun, Dec 13, 2015 at 9:10 PM, Casey McGinley <notifications@github.com>
wrote:

> In main.py
> <https://github.com/toto-build-project/toto-build-and-test/pull/15#discussion_r47456565>
> :
>
> > -  # Setup the command line parser
> > -  args = get_command_line_args()
> > +  commands = []
> > +  # Setup the command policy here.  The user can either pass
> > +  # a file in from the commandline (command_policy option),
> > +  # otherwise pass a single command line entry.
> > +  if args.command_policy:
> > +    # Begin processing the commands from the command policy.
> > +    command_policy_filepath = args.command_policy
> > +    commandList = tuf.util.load_json_file(command_policy_filepath)
> > +    commands = commandList["commands"]
> > +  else:
> > +    # The command is read from the commandline (COMMAND var).
> > +    # We will autoname the file generic_metadata.json.
> > +    tmp_commands = [[args.command, None, "generic", args.input]]
> > +    commands = json.dumps(tmp_commands)
>
> Well I'm not sure we should merge this into master yet. It's definitely
> something to demo and present on Tuesday, but since Prof. Cappos and other
> are currently using our main branch, I don't think we should merge this
> feature yet for fear of affecting stability. I think this branch should be
> kept on hold until the feature is complete, namely that we have a
> client-side script to verify and a method for tracking the tarball hashes
>
> Does that make sense? What do you guys think @ceinfo
> <https://github.com/ceinfo>
> @WilsonLiCode <https://github.com/WilsonLiCode>
> @Justinvalcarcel <https://github.com/Justinvalcarcel>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/toto-build-project/toto-build-and-test/pull/15/files#r47456565>
> .
>